provider/gateway/router.go

```diff
@@ -116,9 +150,152 @@ func leaseServiceStatusHandler(log log.Logger, cclient cluster.ReadClient) http.
 }
 
 func leaseServiceLogsHandler(log log.Logger, cclient cluster.ReadClient) http.HandlerFunc {
-	return func(w http.ResponseWriter, req *http.Request) {
-		http.Error(w, "unimplemented", http.StatusNotImplemented)
+	return func(w http.ResponseWriter, r *http.Request) {
+		service := mux.Vars(r)["serviceName"]
+		if service == "" {
+			http.Error(w, "invalid request", http.StatusBadRequest)
+			return
+		}
+
+		upgrader := websocket.Upgrader{
+			ReadBufferSize:  1024,
+			WriteBufferSize: 1024,
+		}
+
+		ws, err := upgrader.Upgrade(w, r, nil)
+		if err != nil {
+			if _, ok := err.(websocket.HandshakeError); !ok {
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+			}
+			return
+		}
+
+		go wsLogWriter(ws, wsLogsConfig{
+			lid:       requestLeaseID(r),
+			service:   service,
+			follow:    requestLogFollow(r),
+			tailLines: requestLogTailLines(r),
+			log:       log,
+			client:    cclient,
+		})
```

Comment:
I think that this handler should block until everything is done.  does `wsLogWriter()` need to be in a `go`?  I think maybe not.